### PR TITLE
Duplicate SimApp into callbacks directory

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #4338

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/callbacks/callbacks_test.go [hunk 1]
```diff
@@ -22,6 +27,25 @@ import (
...
+
+// GetSimApp returns the duplicated SimApp from within the callbacks directory.
+// This must be used instead of chain.GetSimApp() for tests within this directory.
+func GetSimApp(chain *ibctesting.TestChain) *simapp.SimApp {
+	app, ok := chain.App.(*simapp.SimApp)
+	if !ok {
+		panic("chain is not a simapp.SimApp")
+	}
+	return app
+}
```
Comment (damiannolan) [inline modules/apps/callbacks/callbacks_test.go:47]: For context we essentially went with the same way the ica-demo repo wired up the testing app.

Links:
- https://github.com/cosmos/interchain-accounts-demo/blob/master/x/inter-tx/keeper/keeper_test.go#L35-L44
- https://github.com/cosmos/interchain-accounts-demo/blob/master/x/inter-tx/keeper/keeper_test.go#L98-L105

**Review by damiannolan — APPROVED** [review]
LGTM! We will do a follow up PR to remove the ibccallbacks dep in ibc-go's simapp as well as the contract keeper in there now that is moved to the callbacks module. 

Then we handle creating the separate go module for callbacks! 🚀 🚀 🚀

**Review by srdtrk — APPROVED** [review]
LGTM. I was wondering if it is possible to somehow keep this up to date with the testing directory in ibc-go rather than just copy pasting (maybe something in the Makefile). Not blocking, just a thought. I guess it doesn't matter as this is a temporary solution anyways.

**Comment by chatton** [comment]
@srdtrk  I think it would not be worth the effort to try and automate the sync somehow. There were quite a few small tweaks to things like import paths, and ideally we can remove this is the near term!

**Pull request merged**